libbeat/common/transport/tlscommon/tls_config.go

```diff
@@ -79,17 +83,25 @@ func (c *TLSConfig) ToConfig() *tls.Config {
 		logp.Warn("SSL/TLS verifications disabled.")
 	}
 
+	// When we are usign the CAsha256 pin to validate the CA used to validate the chain
+	// we add a custom callback.
+	var verifyPeerCertFn verifyPeerCertFunc
+	if len(c.CASha256) > 0 {
+		verifyPeerCertFn = MakeCAPinCallback(c.CASha256)
+	}
+
 	return &tls.Config{
-		MinVersion:         minVersion,
-		MaxVersion:         maxVersion,
-		Certificates:       c.Certificates,
-		RootCAs:            c.RootCAs,
-		ClientCAs:          c.ClientCAs,
-		InsecureSkipVerify: insecure,
-		CipherSuites:       c.CipherSuites,
-		CurvePreferences:   c.CurvePreferences,
-		Renegotiation:      c.Renegotiation,
-		ClientAuth:         c.ClientAuth,
+		MinVersion:            minVersion,
+		MaxVersion:            maxVersion,
+		Certificates:          c.Certificates,
+		RootCAs:               c.RootCAs,
+		ClientCAs:             c.ClientCAs,
+		InsecureSkipVerify:    insecure,
+		CipherSuites:          c.CipherSuites,
+		CurvePreferences:      c.CurvePreferences,
+		Renegotiation:         c.Renegotiation,
+		ClientAuth:            c.ClientAuth,
+		VerifyPeerCertificate: verifyPeerCertFn,
```

Comment:
the callback (the way we use it) requires `insecure` to be false. Otherwise the callback will get an empty chain. This is fine, but we should document somewhere (code and or docs), that pinning adds an additional check to normal verification, but is no replacement for e.g. host checks.

Reply:
I have added a comment in both the go code and the user documentation.